### PR TITLE
Add support of new IR pattern to transOCLBuiltinFromVariable

### PR DESCRIPTION
**Pull request by Fznamznon**
Some LLVM optimizations started generation of ascast -> gep -> load
sequence, this patch adds support for it.

### CODE REVIEW

**Comment by Fznamznon** [comment]
@AlexeySotkin , @MrSidims , please take a look.

File: lib/SPIRV/SPIRVReader.cpp [hunk 1]
```diff
@@ -363,10 +381,27 @@ bool SPIRVToLLVM::transOCLBuiltinFromVariable(GlobalVariable *GV,
...
+      Value *Ptr = LD->getPointerOperand();
+
+      if (isa<FixedVectorType>(Ptr->getType()->getPointerElementType())) {
+        LD->replaceAllUsesWith(Vectors.back());
+      } else {
+        auto *GEP = dyn_cast<GetElementPtrInst>(Ptr);
+        assert(GEP && "Unexpected pattern!");
+        assert(GEP->getNumIndices() == 2 && "Unexpected pattern!");
+        Value *Idx = (Value *)GEP->getOperand(2);
+        Value *Vec = (Value *)Vectors.back();
```
Comment (AlexeySotkin) [inline lib/SPIRV/SPIRVReader.cpp:394]:
```suggestion
        auto *Idx = static_cast<Value *>(GEP->getOperand(2));
        auto *Vec = static_cast<Value *>(Vectors.back());
```

**Review by AlexeySotkin — APPROVED** [review]

File: lib/SPIRV/SPIRVReader.cpp (hunk 1 above)
Comment (Fznamznon) [reply lib/SPIRV/SPIRVReader.cpp:394]: Ah, it seems we don't.

**Review by AlexeySotkin — APPROVED** [review]

File: lib/SPIRV/SPIRVReader.cpp [hunk 2]
```diff
@@ -300,7 +314,8 @@ bool SPIRVToLLVM::transOCLBuiltinFromVariable(GlobalVariable *GV,
   } else {
     FuncName = std::string(GV->getName());
   }
-  Type *ReturnTy = GV->getType()->getPointerElementType();
+  Type *GVTy = GV->getType()->getPointerElementType();
+  Type *ReturnTy = GVTy;
```
Comment (MrSidims) [inline lib/SPIRV/SPIRVReader.cpp:318]: This split is done just to improve code readability, correct? I don't see any GVTy/ReturnTy modifications below.

Comment (Fznamznon) [reply lib/SPIRV/SPIRVReader.cpp:318]: No, not only for this. ReturnTy is re-initialized later in the code. I needed to keep original type of the global variable.

Comment (MrSidims) [reply lib/SPIRV/SPIRVReader.cpp:318]: Ah, it's line 325, I've missed it.

**Review by MrSidims — APPROVED** [review]

**Pull request merged**